gdx/src/com/badlogic/gdx/graphics/g2d/Sprite.java

```diff
@@ -338,45 +339,50 @@ public void translate (float xAmount, float yAmount) {
 	/** Sets the color used to tint this sprite. Default is {@link Color#WHITE}. */
 	public void setColor (Color tint) {
 		color.set(tint);
-		float color = tint.toFloatBits();
+		packedColor = tint.toFloatBits();
 		float[] vertices = this.vertices;
-		vertices[C1] = color;
-		vertices[C2] = color;
-		vertices[C3] = color;
-		vertices[C4] = color;
+		vertices[C1] = packedColor;
+		vertices[C2] = packedColor;
+		vertices[C3] = packedColor;
+		vertices[C4] = packedColor;
 	}
 
 	/** Sets the alpha portion of the color used to tint this sprite. */
 	public void setAlpha (float a) {
-		color.a = a;
-		float color = this.color.toFloatBits();
-		vertices[C1] = color;
-		vertices[C2] = color;
-		vertices[C3] = color;
-		vertices[C4] = color;
+		if (color.a != a) {
```

Comment:
This seems OK. Sprite `draw` with alpha modulation uses `setAlpha` but could be more efficient instead.

Reply:
It's not an issue if the == comparison is different and they pack the same float, just misses the optmization on these corner cases. Problem is when the opposite happens.

** Sorry for posting multiple times, the answer was going to the wrong place **